File: Dataset/Leetcode/valid/14/145.java
 class Solution {
    public String XXX(String[] strs) {
        if(strs.length < 1) return "";
        String com = strs[0];
        for(int i = 1; i < strs.length; i++){
            int len = Math.min(strs[i].length(), com.length());
            int j;
            for(j = 0; j < len; j++)
                if(strs[i].charAt(j) != com.charAt(j)) break;
            if(j == 0) return "";
            com = com.substring(0, j);
        }
        return com;
    }
}

File: Dataset/Leetcode/valid/70/145.cpp
class Solution {
public:
    int XXX(int n) {
        
        //dp[n]表示n道楼梯有几种爬的方法
        if(n == 1){
            return 1;
        }
        else if(n == 2){
            return 2;
        }
        else{
            int dp[n+1];
            dp[0] = 0;
            dp[1] = 1;
            dp[2] = 2;
            for(int i = 3;i <= n;++i){
                dp[i] = dp[i-1]+dp[i-2];//经过一步到第i道楼梯与经过两步到i道楼梯的方法数之和
            }
            return dp[n];
        }
    }
};

